src/wallet/wallet.cpp

```diff
@@ -6750,6 +6757,117 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
+    auto recoverPath = GetOptionalArg("-recoverwallet");
+    if (recoverPath.has_value()) {
+        if (!fFirstRun) {
+            return UIError(_("Cannot recover over an existing wallet file."));
+        } else if (fs::exists(recoverPath.value())) {
```

Comment:
```suggestion
        }
        if (!fs::exists(recoverPath.value())) {
            return UIError(_("Wallet recover file not found."));
        }
```
and then unindent and remove the final else clause.